python-frontend/src/main/java/org/sonar/python/semantic/ClassSymbolImpl.java

```diff
@@ -198,11 +198,21 @@ public ClassSymbolImpl copyWithoutUsages() {
     return copiedClassSymbol;
   }
 
+  @Override
+  public List<String> superClassesFqn() {
+    return superClassesFqns;
+  }
+
+  @Override
+  public boolean isStub() {
+    return !hasAlreadyReadSuperClasses && superClasses.isEmpty() && !superClassesFqns.isEmpty();
```

Comment:
When I said this condition was a proxy, I meant that it was the condition used to check in `Typeshed` for the class, but I think it's a bit of a stretch to have it in a dedicated method like this.

If we go for this, maybe we can actually set the `isStub` property when we create any `ClassSymbol` from Typeshed (like we do for `FunctionSymbol`). Or, to keep things simple, we could have a method on the `ClassSymbolImpl` that returns this boolean, with a name such as `shouldSearchHierarchyInTypeshed`.